src/modules/Builder/components/Configuration/components/Main/Main.jsx

```diff
@@ -89,58 +49,79 @@ export default class Main extends Component {
             hostNameValid = false;
         }
 
-        buttonStateCallback("mainHostName", hostNameValid);
+        updateValidStatesAction({ hostNameValid });
+        updateBuildStageAvailableAction(isBuildStageAvailable(validStates));
 
         return hostNameValid;
-    }
-
-    syncConfigParams() {
-        const { hostName, currentTimeZone } = this.state;
-        this.blackmagic.syncConfigurationParams(
-            {
-                hostname: hostName,
-                timezone: currentTimeZone,
-            },
-        );
-    }
-
-    render() {
-        const { hostName, timeZones } = this.state;
-        const selectTimeZones = Main.formatTimeZones(timeZones);
-        return (
-            <Card>
-                <Card.Body>
-                    <div className="configuration-main">
-                        <Form.Group>
-                            <Form.Label>Host name</Form.Label>
-                            <Input
-                                id="hostNameId"
-                                name="hostName"
-                                type="text"
-                                value={hostName}
-                                onChange={this.onHostNameChange}
-                                validationFunc={this.hostNameValidor}
-                            />
-                            <div className="error invalid-feedback">
-                                Host name must be between 1 and 253 characters of Latin letters,
-                                numbers or a dot character
-                            </div>
-                        </Form.Group>
-                        <SelectSearch
-                            ref={this.timeZoneRef}
-                            id="timeZones"
-                            key="time-zone"
-                            label="Time zone"
-                            options={selectTimeZones}
-                            onChange={this.onTimeZoneChange}
+    };
+
+    return (
+        <Card>
+            <Card.Body>
+                <Form>
+                    <Form.Group>
+                        <Form.Label>Host name</Form.Label>
+                        <Form.Control
+                            as={Input}
```

Comment:
Do not use this component, it is deprecated. See how field updates and validation are done in [Profile](https://github.com/tolstoyevsky/cusdeb-web-client/blob/master/src/modules/UserSettings/components/Profile/Profile.jsx) component.